strapi/gcp-deployment/README.md

```diff
@@ -1,3 +1,44 @@
+# Notes on Strapi deployment
+
+## Deploying GCP infrastructure with Terraform
+
+### Setting up
+
+Ensure that `terraform` and `gcloud` are installed.
+
+Set your desired database username and password in `secret.tfvars`.
+
+If you already have a remote terraform state on GCS ready for use (as defined in `backend.tf`),
+simply run `terraform init` to set up terraform.
+Otherwise, set up terraform from scratch as follows:
+1. Comment out `backend.tf`
+2. Run `terraform init`: this initially sets up terraform with a local state
+3. Run `npm run terraform-deploy-init`: this will deploy a GCS bucket to hold your remote terraform state
+4. Uncomment `backend.tf` and replace `bucket` with the name of the created GCS bucket
+5. Run `terraform init` again and enter "yes" when prompted to copy pre-existing state: this will set up terraform to switch to a remote state
+   - If it fails with an error saying that the bucket does not exist, run `gcloud auth application-default login`
+6. (Optional) Delete your local copies of `terraform.tfstate` and `terraform.tfstate.backup`
+
+### Deployment
+
+Run `npm run terraform-deploy` to deploy the GCP resources for Strapi using terraform.
+This may take about 20 minutes.
+
+## Deploying to App Engine
+
+Ensure that `gcloud` is installed and that you are authenticated via `gcloud auth login`.
+Ensure that your Cloud Build service account has the appropriate IAM permissions for deployment by following
+[this section](https://cloud.google.com/build/docs/deploying-builds/deploy-appengine#grant-permissions) of the App Engine deployment guide.
```

Comment:
This is something we should configure in Terraform, e.g. add an IAM binding to the Cloud Build service account via https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/google_project_iam#google_project_iam_member.

IIRC the Cloud Build service account isn't a predefined resource in the tf gcloud provider, but it has a well-defined name, so we could have a

```
data "google_service_account" "service_account" {
  account_id   = "${var.project_id}whatever_the_email_scheme_is"
}
```

Note that this is a `data` not a `resource`, because we are referencing something that already exists and not something we want to be managed via terraform.